index.d.ts

```diff
@@ -0,0 +1,39 @@
+interface Options {
+	/**
+	 * Throw an error when called more than once.
+	 *
+	 * @default false
+	 */
+	throw?: boolean;
+}
+
+/**
+ * Ensure a function is only called once. When called multiple times it will return the return value from the first call.
+ *
+ * @param fn - Function that should only be called once.
+ * @returns A function that only calls `fn` once.
+ */
+export default function oneTime<ArgumentsType extends unknown[], ReturnType>(
+	fn: (...arguments: ArgumentsType) => ReturnType,
+	options?: Options
+): (...arguments: ArgumentsType) => ReturnType;
+
+/**
+ * Get the number of times `fn` has been called.
+ *
+ * @param fn - Function to get call count from.
+ * @returns A number representing how many times `fn` has been called.
+ *
+ * @example
+ *
+ * const foo = onetime(() => {});
+ * foo();
+ * foo();
+ * foo();
+ *
+ * console.log(onetime.callCount(foo));
+ * //=> 3
+ */
+export function callCount(
+	fn: (...arguments: unknown[]) => unknown
+): number | undefined;
```

Comment:
Thanks to the types here, I just realized this might not have the optimal behavior. Do you think the JS implementation should always return a `number`? So if it's an arbitrary function the callCount should be `0`? That kinda makes sense to me.

Reply:
Always returning a number sounds reasonable to me. But there must be a way to signal that a passed in function is actually a onetime function, otherwise it will lead to irritations.

If it was my module, I'd throw an error in case an arbitrary fuction (not onetime) is passed in.